doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -1701,131 +1592,183 @@ String notations
-                        : tactic4 | tactic5
+       .. todo: "Tactic Notation constr := idtac" gives a nice message, would be good to show
+          that message for the "foo" example above.
 
-.. cmd:: Tactic Notation {? (at level @num)} {+ @prod_item} := @tactic.
+     :token:`string`
+       represents a literal value in the notation
 
-   A tactic notation extends the parser and pretty-printer of tactics with a new
-   rule made of the list of production items. It then evaluates into the
-   tactic expression ``tactic``. For simple tactics, it is recommended to use
-   a terminal symbol, i.e. a string, for the first production item. The
-   tactic level indicates the parsing precedence of the tactic notation.
-   This information is particularly relevant for notations of tacticals.
-   Levels 0 to 5 are available (default is 5).
+     :n:`@ident`
+       is the name of a grammar nonterminal listed in the table below.  In a few cases,
+       to maintain backward compatibility, the name differs from the nonterminal name
+       used elsewhere in the documentation.
+
+     :n:`( @ident__parm {? , @string__s } )`
+       :n:`@ident__parm` is the parameter name associated with :n:`@ident`.   The :n:`@string__s`
+       is the separator string to use when :n:`@ident` specifies a list with separators
+       (i.e. :n:`@ident` ends with `_list_sep`).
+
+     :n:`@ltac_expr`
+       The tactic expression to substitute for the notation.  :n:`@ident__parm`
+       tokens appearing in :n:`@ltac_expr` are substituted with the associated
+       nonterminal value.
+
+   For example, the following command defines a notation with a single parameter `x`.
+
+   .. coqtop:: in
 
-   .. cmd:: Print Grammar tactic
+      Tactic Notation "destruct_with_eqn" constr(x) := destruct x eqn:?.
 
-      To know the parsing precedences of the existing tacticals, use the command
-      ``Print Grammar tactic``.
+   For a complex example, examine the 16 `Tactic Notation "setoid_replace"`\s
+   defined in :file:`$COQLIB/theories/Classes/SetoidTactics.v`, which are designed
+   to accept any subset of 4 optional parameters.
 
-   Each type of tactic argument has a specific semantic regarding how it
-   is parsed and how it is interpreted. The semantic is described in the
-   following table. The last command gives examples of tactics which use
-   the corresponding kind of argument.
+   The nonterminals that can specified in the tactic notation are:
+
+   .. todo PR At the moment I splice uconstr.  Should I bring it back?
+      An untyped term is one that doesn't get typechecked??
+
+   .. todo 'open_constr' appears to be another possible value based on the
+      the message from "Tactic Notation open_constr := idtac".
+      Also (at least) "ref", "string", "preident", "int" and "ssrpatternarg".
+      (from reading .v files).
+      Looks like any string passed to "make0" in the code is valid.  But do
+      we want to support all these?
```

Comment:
I think we should add `string`, `open_constr`, maybe stuff from ssr ( we can ask @coq/ssreflect-maintainers ).  See also https://github.com/coq/coq/pull/9122.  I think we should not add duplicates/aliases, and instead remove the undocumented overloading as in https://github.com/coq/coq/pull/9122.  I think we should eventually document all the supported things, but we might want to tidy up some things before supporting them.

Reply:
Will consider in future PRs.  Added a link to your comment in the todo.